### ISSUE TITLE
add ADCIRC 55.01 with gfortran (for version 10.x and those less than 10)

### ISSUE THREAD

**Issue by wwlwpd** [StormSurgeLive/asgs#707]
This is to support the GCC stack that is present on rostam, it be useful for Debian/Ubuntu based installations.

**Comment by jasonfleming** [comment]
When trying to build ADCIRC v55.01 on my desktop Ubuntu 20.04 just now with `build adcirc`, the build failed immediately with the following error message:
```
Build command contained in file, /home/jason/work/adcirc-cg-v55.01-gfortran/asgs-build.sh

Proceed to build? [yes] 

make FOR=ifort FFLAGS="-O2 -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 " \
        FFLAGS90="-O2 -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293 " unhcat.exe
make[1]: Entering directory '/home/jason/work/adcirc-cg-v55.01-gfortran/thirdparty/swan'
ifort HottifySWAN.f90 -c -O2 -W0 -assume byterecl -traceback -diag-disable 8290 -diag-disable 8291 -diag-disable 8293  
make[1]: ifort: Command not found
make[1]: *** [Makefile:214: HottifySWAN.o] Error 127
make[1]: Leaving directory '/home/jason/work/adcirc-cg-v55.01-gfortran/thirdparty/swan'
make: *** [Makefile:188: unhcat] Error 2
build failed (2)
```
I'm on my `issue-718` branch, so this may have already been fixed in master. Just noting it here to capture it.

**Comment by jasonfleming** [comment]
Compiling v53release fails when attempting to build `adcprep` as follows:
```
 icc -I . -I /home/jason/work/adcirc-cg-v53release-gfortran/prep -O2 -DLINUX -I/home/jason/work/opt/default/include  -c /home/jason/work/adcirc-cg-v53release-gfortran/prep/mkdir.c -o /home/jason/work/adcirc-cg-v53release-gfortran/work/odir4/mkdir.o
/bin/bash: icc: command not found
make[1]: *** [makefile:443: /home/jason/work/adcirc-cg-v53release-gfortran/work/odir4/mkdir.o] Error 127
make[1]: Leaving directory '/home/jason/work/adcirc-cg-v53release-gfortran/work'
make: *** [makefile:480: padcirc] Error 2
build failed (2)
```
Again, this is on my Ubuntu 20.04 desktop on the `issue-718` branch. It may have already been fixed in master. I will build ADCIRC manually as a workaround for testing the `issue-718` branch improvements.

**Comment by jasonfleming** [comment]
Had a similar issue when trying to build manually:
```
gfortran -c -I . -I /home/jason/work/adcirc-cg-v53release-gfortran/prep -O2 -ffixed-line-length-none  -DADCNETCDF -I/home/jason/work/opt/default/include -DHAVE_NETCDF4 -DNETCDF_CAN_DEFLATE -DREAL8 -DLINUX -DCSCA -I /home/jason/work/adcirc-cg-v53release-gfortran/work/odir3/ -o /home/jason/work/adcirc-cg-v53release-gfortran/work/odir3/read_input.o  /home/jason/work/adcirc-cg-v53release-gfortran/src/read_input.F
if [ "`echo *.mod`" != '*.mod' ]; then mv *.mod /home/jason/work/adcirc-cg-v53release-gfortran/work/odir3/; fi
icc -I . -I /home/jason/work/adcirc-cg-v53release-gfortran/prep -O2 -DLINUX -I/home/jason/work/opt/default/include  -c /home/jason/work/adcirc-cg-v53release-gfortran/prep/mkdir.c -o /home/jason/work/adcirc-cg-v53release-gfortran/work/odir3/mkdir.o
/bin/bash: icc: command not found
make[1]: *** [makefile:443: /home/jason/work/adcirc-cg-v53release-gfortran/work/odir3/mkdir.o] Error 127
make[1]: Leaving directory '/home/jason/work/adcirc-cg-v53release-gfortran/work'
make: *** [makefile:478: adcirc] Error 2
```

**Comment by jasonfleming** [comment]
@wwlwpd I was able to fix this by commenting out an extra `endif` on line 298 of the v53release `cmplrflags.mk` and then adding the missing `endif` after the `circleci` block. Looking at the v55.01 `cmplrflags.mk`, it looks like there may be multiple extraneous `endif`s  near line 298, but it does not need an extra one after the `circleci` block. I wonder if we should maintain our own cut-down `cmplrlfags.mk` rather than patching the upstream one. Open to any ideas.

**Comment by wwlwpd** [comment]
I believe this was fixed in #1037. Please re-open this issue or create a new one if that is not correct.
